packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -1859,25 +1860,48 @@ describe('packages/date-picker', () => {
 
                       test('error state stays after menu is closed', async () => {
                         const result = renderDatePicker({
-                          value: newUTC(2020, Month.February, 29),
+                          value: newUTC(2020, Month.March, 31),
                         });
                         const input = getRelevantInput(result);
                         userEvent.click(input);
                         userEvent.keyboard('{arrowdown}');
+                        const { menuContainerEl } =
+                          await result.findMenuElements();
                         userEvent.click(result.container.parentElement!);
-                        await waitFor(() => {
-                          const errorElement = result.queryByTestId(
-                            'lg-form_field-error_message',
-                          );
-                          expect(errorElement).toBeInTheDocument();
-                        });
+
+                        await waitForElementToBeRemoved(menuContainerEl);
+                        const errorElement = result.queryByTestId(
+                          'lg-form_field-error_message',
+                        );
+                        expect(errorElement).toBeInTheDocument();
                       });
 
                       break;
                     }
 
                     case 'day': {
-                      // There is no case where decrementing a day results in an invalid date value
+                      test('changing date rolls over to number of days-in-month', async () => {
+                        const result = renderDatePicker({
+                          value: newUTC(2020, Month.February, 1),
+                        });
+                        const input = getRelevantInput(result);
```

Comment:
What input is this?

Reply:
Whichever input is being tested by the `.each` (L1359)

Would it be more readable to have the signature `getRelevantInput(segment, renderResult)`?

Reply:
oh i see, the `.each` makes sense. So if the segment is `year`, pressing `arrowdown` would change the year to 2019?